tests/test_main.py

```diff
@@ -337,3 +337,55 @@ class Config:
     with pytest.raises(ValueError) as exc_info:
         m.b = 11
     assert '"TestModel" object has no field "b"' in str(exc_info)
+
+
+class ValidateAssignmentModel(BaseModel):
+    a: int = 2
+    b: constr(min_length=1)
+    c: constr(min_length=a)
+
+    class Config:
+        validate_assignment = True
+
+
+def test_validating_assignment():
```

Comment:
Unit tests are supposed to test one thing, so if the test fails you know what part of the code has gone wrong.

This isn't always simple, but we can do better here. Split this into `test_validating_assignment_pass` and `test_validating_assignment_fail`.